sdks/java/io/kafka/src/test/java/org/apache/beam/sdk/io/kafka/KafkaIOTest.java

```diff
@@ -691,6 +710,24 @@ public void testSinkDisplayData() {
     assertThat(displayData, hasDisplayItem("retries", 3));
   }
 
+  @Test
+  public void testInferKeyCoder() {
+    CoderRegistry registry = new CoderRegistry();
+    registry.registerStandardCoders();
+
+    assertTrue(KafkaIO.getOrInferCoder(registry, LongDeserializer.class, null)
```

Comment:
A remark here: most of the tests were using `BigEndianLongCoder`, but the registry infers `VarLongCoder` by default for `Long` instead. I do not believe this is an issue, but bear this in mind.